pkg/agent/controller/agent.go

```diff
@@ -273,7 +273,7 @@ func (a *Controller) serviceExportToServiceImport(obj runtime.Object, numRequeue
 			invalidServiceType, fmt.Sprintf("Service of type %v not supported", svc.Spec.Type))
 		klog.Errorf("Service type %q not supported", svc.Spec.Type)
 
-		return nil, true
+		return nil, false
```

Comment:
Why changing this to false?

Reply:
serviceToRemoteServiceImport can cover service change from not supported type to supported type case, by call a.serviceExportToServiceImport.

Reply:
It should. We should add unit test cases to cover the new changes. I can do that in a follow-up PR.